_posts/2023-08-15-securing-wildfly-apps-saml-openshift.adoc

```diff
@@ -0,0 +1,382 @@
+----
+
+== Download and Edit the Keycloak Adapter Configuration File
+
+. From your *simple-webapp-saml* client in the Keycloak Admin Console, click on the *Action* dropdown in the top right corner
+and select *Download Adapter Config*.
+
+. For the *Format option*, select the *Keycloak SAML Adapter keycloak-saml.xml*, and download the file and place it in the example
+application's `WEB-INF` directory, i.e., place the `keycloak-saml.xml` file in
+`/PATH/TO/ELYTRON/EXAMPLES/simple-webapp-saml/src/main/webapp/WEB-INF`.
+
+. Update the `keycloak-saml.xml` file as follows:
+
+* Set the *SP* *entityID* to *"simple-webapp-saml"*
+* Set the *SP* *logoutPage* to *"/simple-webapp-saml"*
+* Replace the *SP* *Keys* configuration with the following configuration, being sure to use the aliases
+and passwords you specified when exporting the SAML keys to the `keystore.jks` file:
++
+[source]
+----
+<Keys>
+    <Key signing="true">
+        <KeyStore password="password" file="/etc/keycloak-saml-secret-volume/keystore.jks">
+            <PrivateKey alias="simple-webapp-saml" password="password"/>
+            <Certificate alias="myrealm"/>
+        </KeyStore>
+    </Key>
+</Keys>
+----
++
+* Push this new file to the `simple-webapp-saml` directory in your `elytron-examples` fork, making
+sure to push the changes to your fork's default branch.
++
+[source]
+----
+cd /PATH/TO/ELYTRON/EXAMPLES/simple-webapp-saml
+git add src/main/webapp/WEB-INF/keycloak-saml.xml
+git commit -m "Added Keycloak adapter deployment descriptor file"
+git push origin main
+----
+
+== Add Helm Configuration
+
+Let's switch to the `charts` directory in our `simple-webapp-saml` example:
+
+[source]
+----
+cd /PATH/TO/ELYTRON/EXAMPLES/simple-webapp-saml/charts
+----
+
+Notice there's a `helm.yaml` file in this directory with the following content:
+
+[source]
+----
+build:
+  uri: https://github.com/YOUR_GITHUB_USERNAME/elytron-examples
+  contextDir: simple-webapp-saml
+deploy:
+  volumes:
+    - name: saml-keystore-volume
+      secret:
+        secretName: simple-webapp-saml-secret
+  volumeMounts:
+    - name: saml-keystore-volume
+      mountPath: /etc/keycloak-saml-secret-volume
+      readOnly: true
+----
+
+Replace *YOUR_GITHUB_USERNAME* in the *build* *uri* with your own GitHub username.
+
+The `helm.yaml` file specifies the Git repository that contains our application's source code.
+
+Because we have modified the application's source code by adding a `keycloak-saml.xml` file, you need
+to set the *build* *uri* in the `helm.yaml` file to point to your own fork.
+
+Notice that our `helm.yaml` file also refers to the OpenShift secret, `simple-webapp-saml-secret` we
+created earlier. This will be used to mount the `keystore.jks` file on our WildFly server pod.
+
+== Deploy the Example Application on WildFly on OpenShift
+
+We can deploy our example application on WildFly on OpenShift using the WildFly Helm Chart:
```

Comment:
just a nitpick s/example application on WildFly/example application to WildFly

Reply:
Fixed, thanks!